x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -323,13 +323,17 @@ protected LogicalPlan doRule(LogicalPlan plan) {
                 return resolveEnrich(p, childrenOutput);
             }
 
-            return plan.transformExpressionsUp(UnresolvedAttribute.class, ua -> resolveAttribute(ua, childrenOutput));
+            return plan.transformExpressionsUp(UnresolvedAttribute.class, ua -> maybeResolveAttribute(ua, childrenOutput));
         }
 
-        private Attribute resolveAttribute(UnresolvedAttribute ua, List<Attribute> childrenOutput) {
+        private Attribute maybeResolveAttribute(UnresolvedAttribute ua, List<Attribute> childrenOutput) {
             if (ua.customMessage()) {
                 return ua;
             }
```

Comment:
In essence, skip this part (quickly returning from resolving an Attribute) and let `eval` have another chance at resolving expressions that use attributes from previous `eval` expressions.